java-manta-it/src/test/java/com/joyent/manta/client/helper/IntegrationTestHelper.java

```diff
@@ -6,49 +13,65 @@
  *
  * @author <a href="https://github.com/1010sachin">Sachin Gupta</a>
+ * @author <a href="https://github.com/nairashwin952013">Ashwin A Nair</a>
  */
-
 public class IntegrationTestHelper {
+    /**
+     * Logger instance.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(IntegrationTestHelper.class);
+
+    private  static final String bucketObjectsSubstring = String.format("%sobjects%s"
+            , SEPARATOR, SEPARATOR);
+
+    private static final String integrationTestBucket = "java-integration-tests";
 
     private static String setupBucketsPath(final ConfigContext config, final String testName) {
-        String bucketPathPrefix = String.format("%s%sbuckets%s%s", config.getMantaHomeDirectory(),
-                MantaClient.SEPARATOR, MantaClient.SEPARATOR,
-                testName.toLowerCase() + UUID.randomUUID());
-        return bucketPathPrefix + MantaClient.SEPARATOR + "objects" + MantaClient.SEPARATOR;
+        final String bucketPathPrefix = String.format("%s%s%s", config.getMantaBucketsDirectory(),
+                SEPARATOR, integrationTestBucket);
+        return bucketPathPrefix + bucketObjectsSubstring + testName;
     }
 
     private static boolean verifyBucketsSupport(final ConfigContext config,
                                                final MantaClient client) throws IOException {
         try {
-            String path = String.format("%s%sbuckets", config.getMantaHomeDirectory(),
-                    MantaClient.SEPARATOR);
+            String path = config.getMantaBucketsDirectory();
             client.options(path);
-        } catch (MantaClientHttpResponseException me) {
-            if (MantaErrorCode.RESOURCE_NOT_FOUND_ERROR.equals(me.getServerCode())) {
-                return false;
+        } catch (MantaClientHttpResponseException e) {
+            if (MantaErrorCode.RESOURCE_NOT_FOUND_ERROR.equals(e.getServerCode())) {
+                LOG.info("Buckets not supported in current Manta: {}", e.getStatusMessage());
             }
+            return false;
         }
         return true;
     }
 
     public static void cleanupTestBucketOrDirectory(final MantaClient client,
                                           final String testPath) throws IOException {
-
-        String bucketObjectsSubstring = String.format("%sobjects%s", MantaClient.SEPARATOR,
-                MantaClient.SEPARATOR);
         if (testPath.contains(bucketObjectsSubstring)) {
-            String bucketPath = testPath.substring(0, testPath.lastIndexOf(bucketObjectsSubstring));
+            final String bucketPath = parseBucketPath(testPath);
             //Delete bucket here
+            try {
+                client.deleteBucket(bucketPath);
+            } catch (MantaClientHttpResponseException e) {
+                if (e.getServerCode().equals(MantaErrorCode.BUCKET_NOT_EMPTY_ERROR)) {
+                    e.setContextValue("Cleanup of non-empty test bucket attempted at:", testPath);
```

Comment:
Context keys are generally in the format of a lower case string with underscores instead of spaces. Perhaps, the key should be something like `bucket_path`.

Reply:
Refactored.